### DISCUSSION TITLE
Mesh leveling issue

### DISCUSSION THREAD

**Discussion by Heros-Tempus** [mriscoc/Ender3V2S1#1032] [Q&A] [answered]
I've been struggling with bed adhesion on my Ender 3 S1 for a while now. It seemed like it was just ignoring the mesh, areas that were measured to be low with the auto built mesh were still printing in air and the high points were still dangerously close to crashing. Then I thought maybe values of the mesh were just too small. I used a mesh editor plugin on octoprint to test my hypothesis. I didn't want to risk making the negative values too small and make it crash, so I only doubled the positive values. It worked, the high spots were no longer printing dangerously close. However, manually editing all those values one at a time is extremely tedious. So I was wondering if there is a way to edit all points in the mesh at once or if I'd be better off just trying to use tape or something to reduce the variance across the bed.

**Comment by bagsofcole** [comment]
I have an Ender 3 V2 NEO w/ CRTouch and everything was fine and dandy getting nearly perfect bed leveling with all points in the green, until I 
A) added the hero me which moves the ABL out about 10mm, or 
B) updated to the newest firmware?
and now the edges of my mesh are all droopy and big flat prints are not level.  
not sure which is the culprit since i did htem both at the same time.  My Neo heatblock stripped so i upgraded to a spider, and that was agood time to do the hero me since it won't accept the new heat sync

**Reply by bagsofcole** [reply]
This may fix your issues....


https://github.com/mriscoc/Ender3V2S1/discussions/1021

Also if you're using UBL and his default gcode for prusa you have to edit it with a different command to activate the stored mesh.

**Reply by Heros-Tempus** [reply]
I tried it and it did not solve my issue. The issue I face isn't that the accuracy of the mesh falls off as I change the resolution of the mesh. The issue is the high areas measured with the probe are higher than measured, and so too are the low areas. Multiplying all the values in the mesh by some scalar value seems to be a workable solution, but fine tuning the scalar is tedious. I don't even know if all the points need the same scalar or if all the points need to be scaled differently. 

And yes, I did configure the starting gcode properly to activate the mesh.

**Reply by mriscoc** [reply, marked as answer]
If there is a scale value that could only happen with wrong steps/mm for the Z axis.

**Reply by Heros-Tempus** [reply]
I just swapped to a different print bed and was making a new mesh when I realized something. Due to how the z steps/mm effects meshes differently whether HS mode is on or off, a comparison between a mesh made with HS mode on and a mesh made with it off can show whether the z steps/mm is set properly. 

If z steps is set improperly then a mesh made with HS off will have a scalar discrepancy because each point is measured relative to the bed. The resulting mesh will either have exaggerated or muted features depending on if the z steps/mm is too high or low.

With HS mode on, on the other hand, each point will have an additive discrepancy because each point is measured relative to the previous points so the effect of the poor calibration compounds with each point measured. The resulting mesh will have points of increasing magnitude, going higher or lower depending on if the z steps/mm is too high or low.

If z steps is set properly (or close to it) the two meshes will closely correspond. 

I'm going to try to write some code that uses this property to automatically adjust the z steps/mm.

**Reply by mriscoc** [reply]
All probe points are measured respect to the latest Z home, with or without HS mode.
